lock-api/src/main/java/com/palantir/lock/client/LeaderElectionReportingTimelockService.java

```diff
@@ -164,6 +170,12 @@ LeaderElectionStatistics getStatisticsAndSetRegistryTo(TaggedMetricRegistry metr
         boolean election = !newLeader.equals(currentLeader);
 
         if (election) {
+            if (!newLeader.equals(leaderId)) {
```

Comment:
When would you expect this to be false when election is true? If my read is correct:
`election = !newLeader.equals(currentLeader)`and `currentLeader = leaderId`.
Is this because of the potential for a race condition?

Reply:
Yup, this is because all the requests that were sent during the leader election will come here, so this is just to filter out excessive logging, as once the leaderId gets updated, the rest will not be logged. Note that we can still race condition into getting more than one log line, but that's fine

Reply:
That's reasonable.